CRM/Financial/BAO/FinancialType.php

```diff
@@ -246,6 +246,38 @@ public static function permissionedFinancialTypes(&$permissions, $descriptions)
   }
 
   /**
+   * Wrapper aroung getAvaliableFinancialTypes to get all including disabled FinancialTypes
+   * @param int|string $action
+   *   the type of action, can be add, view, edit, delete
+   * @param bool $resetCache
+   *   load values from static cache
+   *
+   * @return array
+   */
+  public static function getAllAvailableFinancialTypes($action = CRM_Core_Action::VIEW, $resetCache = FALSE) {
+    // Flush pseudoconstant cache
+    CRM_Contribute_PseudoConstant::flush('financialType');
+    $thisIsAUselessVariableButSolvesPHPError = NULL;
```

Comment:
Note that this is needed because PHP7.1 especially will only allow for variables to be passed by reference